modules/HumanStateVisualizer/src/main.cpp

```diff
@@ -0,0 +1,259 @@
+                return EXIT_FAILURE;
+            }
+            yWarning() << LogPrefix << "joint [ " << jointName << " ] not found in the visualized model, "
+                       << "the joint will be ignored.";
+        }
+    }
+
+    // initialize state variables for visualization
+    iDynTree::Transform wHb = iDynTree::Transform::Identity();
+    iDynTree::VectorDynSize joints(viz.modelViz("human").model().getNrOfDOFs());
+    joints.zero();
+
+    // start Visualization
+    std::chrono::steady_clock::time_point now = std::chrono::steady_clock::now();
+    std::chrono::steady_clock::time_point lastViz = std::chrono::steady_clock::now();
+
+    long minimumMicroSecViz = std::round(1e6 / (double) maxVisualizationFPS);
+
+    while(viz.run())
+    {
+        now = std::chrono::steady_clock::now();
+        if (std::chrono::duration_cast<std::chrono::microseconds>(now - lastViz).count() < minimumMicroSecViz)
+        {
+            std::this_thread::sleep_for(std::chrono::milliseconds(1));
+            continue;
+        }
+
+        // read values from iHumanState
+        std::array<double, 3> basePositionInterface = iHumanState->getBasePosition();
+        std::array<double, 4> baseOrientationInterface = iHumanState->getBaseOrientation();
+        std::vector<double> jointPositionsInterface = iHumanState->getJointPositions();
+        std::vector<std::string> jointNames = iHumanState->getJointNames();
+        std::string baseName = iHumanState->getBaseName();
+
+        iDynTree::Vector4 baseOrientationQuaternion;
+        baseOrientationQuaternion.setVal(0, baseOrientationInterface.at(0));
+        baseOrientationQuaternion.setVal(1, baseOrientationInterface.at(1));
+        baseOrientationQuaternion.setVal(2, baseOrientationInterface.at(2));
+        baseOrientationQuaternion.setVal(3, baseOrientationInterface.at(3));
+
+        iDynTree::Position basePosition;
+        basePosition.setVal(0, basePositionInterface.at(0));
+        basePosition.setVal(1, basePositionInterface.at(1));
+        basePosition.setVal(2, basePositionInterface.at(2));
+
+        wHb.setRotation(iDynTree::Rotation::RotationFromQuaternion(baseOrientationQuaternion));
+        wHb.setPosition(basePosition);
+
+        for (size_t iHumanInterfaceIdx = 0; iHumanInterfaceIdx < jointNames.size(); iHumanInterfaceIdx++)
+        {
+            std::string jointName = jointNames.at(iHumanInterfaceIdx);
+            double jointVal = jointPositionsInterface.at(iHumanInterfaceIdx);
+
+            iDynTree::JointIndex jointIndex = viz.modelViz("human").model().getJointIndex(jointName);
+            if (jointIndex != iDynTree::JOINT_INVALID_INDEX)
+            {
+                joints.setVal(jointIndex, jointVal);
+            }
+        }
+
+        // follow the desired link with the camera
+        if ( !useFixedCamera )
+        {
+            viz.camera().setPosition(basePosition + cameraDeltaPosition);
+            viz.camera().setTarget(basePosition);
+        }

```

Comment:
I would be curious to see if it possible to have the camera fixed and also have the mouse control :thinking:

Reply:
I think it may be possible if we update the ``cameraDeltaPosition`` by reading the position of the camera and of the target directly from the visualizer. But unfortunately, this is not possible yet, but it should be pretty easy to add.

Reply:
> I think it may be possible if we update the cameraDeltaPosition by reading the position of the camera and of the target directly from the visualizer after calling the draw method.

Yes, I think that is the way to combine the model-following camera and the mouse control. I could have a look at the missing methods after integrating the visualization++ features.

Reply:
indeed it worked (see https://github.com/robotology/human-dynamics-estimation/pull/233)